lib/common/options.js

```diff
@@ -161,6 +161,10 @@ Options.from = function (_options) {
     vueConfig.performance = options.dev
   }
 
+  // merge custom env with variables
+  const eligibleEnvVariables = _.pick(process.env, Object.keys(process.env).filter(k => k.startsWith('NUXT_ENV_')))
+  options.env = Object.assign(eligibleEnvVariables, options.env)
```

Comment:
Shouldn't env variables override `options.env`? Maybe it would be better to use `Object.assign({}, options.env, eligibleEnvVariables)`

Reply:
I wasn't certain which behavior would be more appropriate. 

On the one hand, it feels like someone "automatically injected" should have an option to get overridden.

On the other hand, env variables are way more "closer" to the system...

Reply:
The last point would be [consistency](https://github.com/nuxt/nuxt.js/pull/3701). I'll change it to the suggested value :relaxed: